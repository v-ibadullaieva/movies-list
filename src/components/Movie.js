import React, { Component } from "react";
import { connect } from "react-redux";
import { Link } from "react-router";
import { fetchMovie } from "../features/movies/actions";

export class Movie extends Component {
  componentDidMount() {
    if (!this.props.movie) {
      this.props.fetchMovie(this.props.params.id);
    }
  }

  render() {
    const { movie } = this.props;

    if (movie) {
      return (
        <div>
          <Link to="/" className="page__link">
            Back to movies
          </Link>
          <li>
            <h3>{movie.title}</h3>
            <p>{movie.year}</p>
            <p>{movie.format}</p>
            <ul>
              {movie.actors.map((actor, i) => <li key={i}>{actor.name}</li>)}
            </ul>
          </li>
        </div>
      );
    } else {
      return null;
    }
  }
}

export default connect(
  (state, props) => ({
    movie: state.app.movies[props.params.id]
  }),
  { fetchMovie }
)(Movie);
