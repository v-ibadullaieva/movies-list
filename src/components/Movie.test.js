import React from "react";
import { Movie } from "./Movie";
import renderer from "react-test-renderer";

const movie = {
	title: "title",
	year: "1998",
	format: "vhs",
	actors: [{ name: "John" }]
};

it("renders without crashing", () => {
	const tree = renderer
		.create(<Movie movie={movie} params={{ id: 99 }} />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
