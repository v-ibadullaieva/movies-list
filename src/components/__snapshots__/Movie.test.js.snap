// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <a
    className="page__link"
    onClick={[Function]}
    style={Object {}}
  >
    Back to movies
  </a>
  <li>
    <h3>
      title
    </h3>
    <p>
      1998
    </p>
    <p>
      vhs
    </p>
    <ul>
      <li>
        John
      </li>
    </ul>
  </li>
</div>
`;
